src/app/campaign/flight/flight-zones-form.component.ts

```diff
@@ -96,20 +105,23 @@ export class FlightZonesFormComponent implements ControlValueAccessor, OnDestroy
   }
 
   writeValue(zones: FlightZone[]) {
+    // don't emit while manipulating FormArray with incoming update
+    this.emitGuard = true;
     // get the correct number of zone fields and patchValue
-    const expectedLength = zones.length || 1;
-    while (this.zones.controls.length > expectedLength) {
+    zones = zones && zones.length ? zones : [{ id: '', url: '' }];
+    while (this.zones.controls.length > zones.length) {
       this.zones.removeAt(this.zones.controls.length - 1);
       this.zones.markAsPristine();
     }
     const addZones = [...zones];
-    while (this.zones.controls.length < expectedLength) {
+    while (this.zones.controls.length < zones.length) {
       // this only adds controls, doesnt reset controls already in form
       this.zones.push(this.flightZoneFormGroup(zones && addZones.pop()));
       this.zones.markAsPristine();
     }
     // patch all of the values onto the FormArray
-    this.zones.patchValue(zones || [], { emitEvent: false });
+    this.zones.patchValue(zones, { emitEvent: false, onlySelf: true });
```

Comment:
So this prevents triggering a form-changed event on the parent form?

Reply:
yeah, I guess that's what onlySelf is for, just update this control and any children. I think in practice, it's not doing anything extra for us here with the emit guard, but best practices and all (once you know better.)